### PR TITLE
Add bundle split capabilities in ModularLoadManagerImpl

### PR DESCRIPTION
**Pull request by bobbeyreese**
### Motivation

`ModularLoadManagerImpl` lacked bundle split capabilities. Additionally, automatic bundle splits reassigned the bundles to the same broker when the bundle could more usefully be reassigned after being split.

### Modifications
Roughly in order of importance:

* Automatic bundle split is now available in `ModularLoadManagerImpl`.
* Automatic bundle split has been altered so that it also unloads the bundle: this allows splits to be reassigned to new brokers.
*  Bundle boundaries are now stored as ZNodes of the same names on ZooKeeper under the local policies. This is necessary because, with automatic bundle splitting without broker affinity, multiple brokers may set the boundaries simultaneously, resulting in inconsistent boundaries if one write overwrites the other. Successful bundle splitting has been observed with as many as 5 bundle splits on the same namespace happening simultaneously.
* `ZooKeeperCache` will now *always* put a watch on a node, even if it did not previously exist. This is done via using the capability of `ZooKeeper.exists` to put a watch on non-existent nodes.
* `BundleSplitStrategy` has been added to allow the injection of automatic bundle splitting behavior.
* `loadBalancerAutoBundleSplitEnabled` is now a dynamic configuration.
* `BundleData` -> `TimeAverageBundleData`
* New class `BundleData` holds non-time average (numBundles/Topics, etc) bundle values
* New class 'MessageData` holds message rate in/out and throughput in/out. 
* Components of `ModularLoadManagerImpl`such as `ModularLoadManagerStrategy` are now exposed the `PulsarService` instance instead of `ServiceConfiguration` to give more information and freedom.

### Result

This PR adds a bundle split mechanism for `ModularLoadManagerImpl` and changes bundle splitting behavior so that bundles which are split are also unloaded, so that the splits may be assigned to new brokers.

### CODE REVIEW

File: pulsar-zookeeper-utils/src/main/java/com/yahoo/pulsar/zookeeper/ZooKeeperCache.java [hunk 1]
```diff
@@ -289,28 +290,38 @@ public Boolean call() throws Exception {
 
             // Broker doesn't restart on global-zk session lost: so handling unexpected exception
             try {
-                this.zkSession.get().getData(path, watcher, (rc, path1, ctx, content, stat) -> {
+                // Put watch on the node no matter what
```
Comment (merlimat) [inline pulsar-zookeeper-utils/src/main/java/com/yahoo/pulsar/zookeeper/ZooKeeperCache.java]: Can you describe this change a bit more? Since it doesn't affect only the bundle splitting in the load manager, it would be better to split it in a separate PR.

Comment (bobbeyreese) [inline pulsar-zookeeper-utils/src/main/java/com/yahoo/pulsar/zookeeper/ZooKeeperCache.java]: Sure! This uses the capability of `ZooKeeper.exists` to put a watch on nonexistent nodes. The absence of this behavior caused an issue in which namespace bundle boundary updates were not being seen across all brokers. I will make a separate PR w/ just the cache change.

**Comment by rdhabalia** [comment]
should we close this PR in favor of #857 ?

**Comment by merlimat** [comment]
Yes closing for #857
